v3 - Chakra-UI v1 & Next.js v10

franky47 opened the pull request:
- [x] Port existing components to Chakra-UI v1 API
- [x] Simplify _app.ts
- [x] ~Introduce _document.ts to add colorMode persistance~
- [x] ~Add shorthands for `Card` layout types: `<Card.Flex>`, `<Card.Stack>` etc..~ -> No need, replaced by the `as` prop
- [x] Add Image component (see #187)
- [ ] Use [`forwardRef`](https://chakra-ui.com/guides/as-prop#option-1-using-forwardref-from-chakra-uireact) for custom components
- [ ] Mark support for `as` in Links as optional
- [ ] Handle route links in menus (#221)

Documentation
- [ ] Migration guide
- [ ] OutgoingLink `rel` options

Closes #187, #221.

franky47 committed:
chore: Update dependencies

Starting Chakra v1 migration

franky47 committed:
chore: Upgrade existing components

BREAKING CHANGE: `SvgBox` renamed `Svg`
BREAKING CHANGE: Replaced `PseudoBox` by `Box` (Chakra-UI v1)

franky47 committed:
chore: Ignore Next stuff

franky47 committed:
chore: Remove containers (now part of Chakra core)

BREAKING CHANGE: Removed `Container`, `StackContainer` and `FlexContainer`.

franky47 committed:
chore: Simplify _app.tsx

BREAKING CHANGE: `enableColorMode` is now part of Chakra (in `theme.config`)
BREAKING CHANGE: `globalCss` is now part of Chakra (in `theme.styles.global`)
BREAKING CHANGE: `getGlobalConfig` is deprecated
and spread around in Chakra's config & global styles.

franky47 committed:
test: Fix links test

Somehow the new Chakra UI styles don't play well with
JSDOM's CSS resolution, and textDecoration is not read
correctly. Runtime tests show that the correct styles are
applied, so we test the logic against another property
(color, which seems to go through).

franky47 committed:
feat: Add basic Image component

franky47 commented on package.json:
```diff
@@ -72,6 +72,9 @@
     "preset": "ts-jest/presets/js-with-ts",
     "testEnvironment": "jsdom",
+    "setupFiles": [
+      "./test/env.js"
```
Note: This is a workaround for Next.js image config not having a default, should be resolved in the next version:
https://github.com/vercel/next.js/discussions/18373#discussioncomment-114212

trevorblades commented:
Stoked for this release! Is there any way you could publish a prerelease version off of this branch? It seems like the repo is set up to auto deploy, but I thought I'd ask anyway. If the answer is no, I'm happy to be patient 😇

franky47 commented:
Hey Trevor, that sounds like a good idea, I'll look into how to do this. If you have experience with semantic-release, any help is welcome!

franky47 committed:
chore: Enable beta release channel

franky47 commented:
@trevorblades you can give it a try: `@47ng/chakra-next@3.0.0-beta.1`, feel free to open an issue if you encounter any problems. Breaking changes are listed in the [changelog](https://github.com/47ng/chakra-next/releases/tag/v3.0.0-beta.1).

_**Note to self:** deploying pre-releases is extremely simple with `semantic-release`: enable CD to run on the `beta` branch, then point that branch to whatever you want to deploy, it will semver up and handle dist-tags automatically. 🚀_

franky47 committed:
chore: Update deps

trevorblades commented:
That's great to hear @franky47! I meant to investigate this, but you got to it before me. Thanks 😄

dependabot-preview[bot] committed:
chore: Enable CI on v3 branch

franky47 committed:
fix: Remove automatic inclusion of Image

Doing so adds `next/image` in the imports tree,
which is enough to automatically opt-in to Next.js
Image Optimization feature, which has drawbacks:
- Breaks static export
- Requires a server / serverless setup
- Requires setting up the Image Optimization config

Buying in the Image component should be explicit,
so imports will be separated.

See #187.

franky47 commented:
Hey @trevorblades, as a heads-up, I'm changing the way the Image component is included.

In `3.0.0-beta.1`, it was included at the top-level with the rest, but this has a major drawback: it automatically opts-in to the Next.js Image Optimization feature, even if no Image component is actually used (simply importing '@47ng/chakra-next' made static exports impossible for example, as Image Optimization requires a server or an image loader).

So I'm moving it out of the top-level, the component can be accessed in `3.0.0-beta.2` the following way:

```ts
import { Image } from '@47ng/chakra-next/dist/components/image'
```

franky47 committed:
chore: Add funding options

franky47 committed:
fix: Explicitly declare path to TS types

This lets npmjs.com know that the project has types,
and shows a helpful TypeScript icon.

franky47 committed:
doc: Update funding

franky47 committed:
fix: Improve accessibility of NavLinks

Source: https://twitter.com/toktoktwan/status/1311567989955801088

franky47 committed:
feat: Add control over rel in outgoing links

franky47 committed:
chore: Update deps

franky47 committed:
test: Fix TypeScript error

franky47 committed:
test: Add link tests

franky47 committed:
fix: Update external link icon

franky47 committed:
chore: Include react-icons in devDeps for build

franky47 committed:
fix: External icon layout

franky47 committed:
fix: Allow custom roles for SVG

franky47 committed:
chore: Migrate to GH-builtin Dependabot

franky47 committed:
chore: Use Docker image for semantic-release

franky47 committed:
chore: Update Next.js & fix missing types

franky47 committed:
chore: Notify of CI failures in Slack

franky47 committed:
chore: Testing new Slack notifications
